### PR TITLE
makefile: add release-src in makefile.

### PR DESCRIPTION
**Pull request by moonming**
### What this PR does / why we need it:

Using this script we can generate the release package more quickly
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -190,3 +194,29 @@ ifeq ("$(wildcard .travis/openwhisk-utilities/scancode/scanCode.py)", "")
 endif
 	.travis/openwhisk-utilities/scancode/scanCode.py --config .travis/ASF-Release.cfg ./
 
+release-src:
```
Comment (membphis) [inline Makefile:197]: add some comment in this way: https://github.com/apache/apisix/pull/2481/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R188

Then we can get help information through the command `make help`.

Comment (moonming) [reply Makefile:197]: It is ony for Apache release manager

File: Makefile [hunk 2]
```diff
@@ -24,6 +24,10 @@ UNAME ?= $(shell uname)
 OR_EXEC ?= $(shell which openresty)
 LUAROCKS_VER ?= $(shell luarocks --version | grep -E -o  "luarocks [0-9]+.")
 
+SHELL := /bin/bash -o pipefail
```
Comment (membphis) [inline Makefile:27]: seems that this line is not necessary. can we delete it?

**Review by membphis — APPROVED** [review]
LGTM, except for minor things.

**Review by juzhiyuan — APPROVED** [review]
LGTM

**Pull request merged**